plugins/camera/camera.h

```diff
@@ -0,0 +1,381 @@
+     */
+    void set_mode_async(Mode mode, const mode_callback_t &callback);
+
+    /**
+     * @brief Getter for camera mode (asynchronous).
+     *
+     * @param callback Function to call with result of request.
+     */
+    void get_mode_async(const mode_callback_t &callback);
+
+    /**
+     * @brief Information about a picture just captured.
+     */
+    struct CaptureInfo {
+        /**
+         * @brief Position type.
+         */
+        struct Position {
+            double latitude_deg; /**< @brief Latitude in degrees (range: -90 to +90) */
+            double longitude_deg; /**< @brief Longitude in degrees (range: -180 to 180) */
+            float absolute_altitude_m; /**< @brief Altitude AMSL (above mean sea level) in metres */
+            float relative_altitude_m; /**< @brief Altitude relative to takeoff altitude in metres */
+        } position; /**< @brief Position of drone/camera when image was captured. */
+
+        /**
+         * @brief Quaternion type.
+         *
+         * All rotations and axis systems follow the right-hand rule.
+         * The Hamilton quaternion product definition is used.
+         * A zero-rotation quaternion is represented by (1,0,0,0).
+         * The quaternion could also be written as w + xi + yj + zk.
+         *
+         * For more info see: https://en.wikipedia.org/wiki/Quaternion
+         */
+        struct Quaternion {
+            float w; /**< @brief Quaternion entry 0 also denoted as a. */
+            float x; /**< @brief Quaternion entry 1 also denoted as b. */
+            float y; /**< @brief Quaternion entry 2 also denoted as c. */
+            float z; /**< @brief Quaternion entry 3 also denoted as d. */
+        } quaternion; /**< @brief Quaternion of camera orientation. */
+
+        uint64_t time_utc_us; /**< @brief timestamp in UTC (since UNIX epoch) in microseconds. */
+        bool success; /**< @brief True if capture was successful. */
+        int index; /**< @brief Zero-based index of this image since armed. */
+        std::string file_url; /**< URL of image taken to download. */
+    };
+
+    /**
+     * @brief Type for video stream settings.
+     *
+     * Application may call set_video_stream_settings() before starting
+     * video streaming to tell Camera server to use these settings during video streaming.
+     *
+     * @sa set_video_stream_settings(), get_video_stream_info(), start_video_streaming().
+     */
+    struct VideoStreamSettings {
+        float frame_rate_hz = 0.f; /**< @brief Frames per second. */
+        uint16_t resolution_h_pix = 0u; /**< @brief Resolution horizontal in pixels. */
+        uint16_t resolution_v_pix = 0u; /**< @brief Resolution verticle in pixels. */
+        uint32_t bit_rate_b_s = 0u; /**< @brief Bit rate in bits per second. */
+        uint16_t rotation_deg = 0u; /**< @brief Video image rotation clockwise (0-359 degrees) */
+        std::string uri {}; /**< @brief Video stream URI */
```

Comment:
Terminate consistently with full stops (across all of these).

BTW, I am not sure this bit of the spec works yet (RTSP stream setting via MAVLink)